### ISSUE TITLE
[unbound-method] should accept a whitelist

### ISSUE THREAD

**Issue by morithon** [typescript-eslint/typescript-eslint#1776]
**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/unbound-method": ["warn", { ignoreStatic: true, whitelist: ["expect"] }]
  }
}
```

```TS
spyOn(el, "setAttribute");
expect(el.setAttribute).not.toHaveBeenCalled();
```

**Expected Result**
For eslint not to throw an error

**Actual Result**
Eslint throws an error

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.15.0` |
| `@typescript-eslint/parser`        | `2.15.0` |
| `TypeScript`                       | `3.7.5` |
| `ESLint`                           | `6.8.0` |
| `node`                             | `10.16.0` |
| `npm`                              | `6.9.0` |

It would be nice to have an ability to define exceptions to the rule using a whitelist option.

**Comment by bradzacher** [comment]
It would probably be better to disable the rule in test files.
https://eslint.org/docs/user-guide/configuring#configuration-based-on-glob-patterns 

If there are cases in your codebase where you are okay with passing unbound methods, then you can also use disable comments.

Whitelists are generally bad because they lack context. This is doubly true for type-aware rules, as they need even more context.
For example, if I whitelist the function `foo`, intending to target the function in file a.ts, but then define a function `foo` in file b.ts, the rule has no way to tell that its not supposed to whitelist the second function. This causes false negatives, and introduces bug vectors into your codebase.
